run_linux.go

```diff
@@ -2733,14 +2741,34 @@ func (b *Builder) getSSHMount(tokens []string, count int, sshsources map[string]
 }
 
 // cleanupRunMounts cleans up run mounts so they only appear in this run.
-func cleanupRunMounts(mountpoint string, artifacts *runMountArtifacts) error {
+func (b *Builder) cleanupRunMounts(context *imagetypes.SystemContext, mountpoint string, artifacts *runMountArtifacts) error {
 	for _, agent := range artifacts.Agents {
 		err := agent.Shutdown()
 		if err != nil {
 			return err
 		}
 	}
 
+	//cleanup any mounted images for this run
+	for _, image := range artifacts.MountedImages {
+		if image != "" {
+			i, err := internalUtil.LookupImage(context, b.store, image)
+			// its not necessary we are going to find image, could be stage as well.
+			if err == nil {
+				// silently try to unmount and do nothing
+				// if image is being used by something else
+				_ = i.Unmount(false)
+			}
+			if errors.Unwrap(errors.Unwrap(err)) == storagetypes.ErrImageUnknown {
```

Comment:
This would be a good place to use `errors.Cause()`, so we wouldn't be depending on how many levels of wrapping the underlying error had gotten, but we should be able to know the correct thing to do for a given name - the logic that gave us that name knew what it was referring to, but it currently doesn't tell us.

Reply:
Switched to `errors.Cause`

> If it's not an image, is it a container? It's a problem that we don't already know what the right thing to do is when we get here.

> but we should be able to know the correct thing to do for a given name - the logic that gave us that name knew what it was referring to, but it currently doesn't tell us.

About this part the idea is that if its `ErrImageUnknown` then either original image is unmounted already or its `stage/container`.
For `stage/container` we can only clean up the bind mount which we did for our `--mount` which we already did but we are not authorized to perform explicit `unmount` of `stage or container` 's mountPoint rootfs since its handled by `stage` or `containers` lifecycle.

I also added this as comment in code.